# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannelParent.cpp
# Commit: abd9cd77f3cb
# Full Hash: abd9cd77f3cb3e6a7734af2b8e4a22de4b931382
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-06-09 15:16:49
# Regressor Bug: 1641737
# File Overlap Count: 1
# Description:
#   Bug 1641737 - P5. Remove unnecessary method. r=mattwoodrow
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78210
# ==============================================================================

diff -r e3bf9a45db6a -r abd9cd77f3cb netwerk/ipc/DocumentChannelParent.cpp
--- a/netwerk/ipc/DocumentChannelParent.cpp	Tue Jun 09 07:23:57 2020 +0000
+++ b/netwerk/ipc/DocumentChannelParent.cpp	Tue Jun 09 07:24:04 2020 +0000
@@ -79,8 +79,7 @@
         self->mDocumentLoadListener = nullptr;
       },
       [self](DocumentLoadListener::OpenPromiseFailedType&& aRejectValue) {
-        if (aRejectValue.mStatus == NS_ERROR_DOCSHELL_DYING) {
-          Unused << self->SendDeleteSelf();
+        if (!self->CanSend()) {
           return;
         }
         Unused << self->SendDisconnectChildListeners(